[ERM UI] Create domain link RMI

gherceg opened the pull request:
## Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
[JIRA Ticket](https://dimagi-dev.atlassian.net/browse/SS-147)

This PR handles making the RMI call from the frontend to create a domain link. The `DomainLink` object itself is quite simple, so this should be sufficient to start linking data models.

I was planning on displaying the `DomainLinkError` exception on the frontend, but thought twice about it from a security perspective. The state it is in right now doesn't make sense, as I would either use the message or not send it in the response at all. Any thoughts?

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
`LINKED_DOMAINS`
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
Will run through QA on parent branch.
### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->
Locally tested on both a newly created link and a reinstated link (one that has been removed but still exists). 
### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

gherceg committed:
basic implementation to create domain link via rmi

gherceg committed:
stickler indentation issue

orangejenny commented on corehq/apps/linked_domain/view_helpers.py:
```diff
@@ -32,6 +32,16 @@
+        'master_domain': link.qualified_master,
+        'remote_base_url': link.remote_base_url,
+        'is_remote': link.is_remote,
+        'last_update': server_to_user_time(link.last_pull, timezone) if link.last_pull else 'Never',
```
translate

orangejenny commented on corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:
```diff
@@ -173,18 +173,18 @@ hqDefine("linked_domain/js/domain_links", [
+                    return item !== viewModel.value();
+                }));
+                self.value(null);
+                self.parent.domain_links.unshift(DomainLink(data.response));
```
Shouldn't this check for the `success` value and display the error message if it's there?

gherceg replied:
Updated to check success, as well as display the error message returned in this case.

orangejenny reviewed (approved):
Just a minor comment on js error handling.

gherceg committed:
address feedback and minor edits

- check success in js response
- translate 'Never'
- refactor link_context to clearer name
- display exception message to user
- minor stickler complaints

orangejenny commented on corehq/apps/linked_domain/static/linked_domain/js/domain_links.js:
```diff
@@ -175,15 +175,19 @@ hqDefine("linked_domain/js/domain_links", [
+                    self.parent.domain_links.unshift(DomainLink(response.domain_link));
+                    self.parent.goToPage(1);
+                } else {
+                    alertUser.alert_user(gettext('Unable to link project spaces. ' + response.message + '\nYou must remove the existing link before creating this new link.'), 'danger');
```
This won't translate because it's dynamic. We often use `_.template` in js similarly to python's `format` to build translate-able strings.

gherceg committed:
use _.template to translate dynamic string

gherceg committed:
pass in timezone object

gherceg committed:
fix views.py import statements
